util/jenkins/primary_keys/check_primary_keys.py

```diff
@@ -0,0 +1,211 @@
+            temp_dict = dict()
+            temp_dict["name"] = instance["DBInstanceIdentifier"]
+            temp_dict["Endpoint"] = instance.get("Endpoint").get("Address")
+            temp_dict["Port"] = instance.get("Port")
+            rds_list.append(temp_dict)
+    return rds_list
+
+
+def check_primary_keys(rds_list, username, password, environment, deploy):
+    """
+    :param rds_list:
+    :param username:
+    :param password:
+
+    :returns:
+         Return list of all tables that cross threshold limit
+              [
+                  {
+                    "name": "string",
+                    "db": "string",
+                    "table": "string",
+                    "size": "string",
+                  }
+              ]
+    """
+    cloudwatch = CwBotoWrapper()
+    metric_name = 'check_primary_keys'
+    namespace = "rds-primary-keys/{}-{}".format(environment, deploy)
+    try:
+        table_list = []
+        metric_data = []
+        for item in rds_list:
+            rds_host_endpoint = item["Endpoint"]
+            rds_port = item["Port"]
+            connection = pymysql.connect(host=rds_host_endpoint,
+                                         port=rds_port,
+                                         user=username,
+                                         password=password)
+            # prepare a cursor object using cursor() method
+            cursor = connection.cursor()
+            # execute SQL query using execute() method.
+            cursor.execute("""
+            SELECT table_schema, table_name, column_name, column_type, auto_increment,max_int,
+            ROUND(auto_increment/max_int*100,2) 
+            AS "USED %" FROM 
+            (SELECT table_schema, table_name, column_name, column_type, auto_increment, 
+            pow(2, case data_type when 'tinyint' then 7 when 'smallint' then 15 when 'mediumint' 
+            then 23 when 'int' then 31 when 'bigint' then 63 end+(column_type like '% unsigned'))-1 as 
+            max_int FROM 
+            information_schema.tables t JOIN information_schema.columns c USING 
+            (table_schema,table_name) WHERE t.table_schema not in 
+            ('mysql','information_schema','performance_schema') and 
+            t.table_type = 'base table' and c.extra = 'auto_increment' 
+            AND t.auto_increment IS NOT NULL order by auto_increment desc limit 10) TMP;
+            """)
+            rds_result = cursor.fetchall()
+            cursor.close()
+            connection.close()
+
+            for tables in rds_result:
+                metric_data.append({
+                    'MetricName': metric_name,
+                    'Dimensions': [{
+                        "Name": item["name"],
+                        "Value": tables[0]
+                    }],
+                    'Value': tables[6] # percentnage of the usage of primary keys
+                })
+
+                if len(metric_data) > 0:
```

Comment:
```suggestion
        if len(metric_data) > 0:
```
This should be run once after the for item loop.